sdk/scheduler/src/main/java/com/mesosphere/sdk/metrics/PlanReporter.java

```diff
@@ -0,0 +1,53 @@
+package com.mesosphere.sdk.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.mesosphere.sdk.scheduler.plan.Plan;
+import com.mesosphere.sdk.scheduler.plan.PlanManager;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This class encapsulates a simple reporter that scrapes plan statuses at a regular interval (5 seconds)
+ * and outputs them to the metrics sink.
+ */
+public class PlanReporter {
+
+    private final List<PlanManager> managers;
+    private final ScheduledExecutorService executor;
+    private final Optional<String> namespace;
+    @VisibleForTesting
+    final AtomicBoolean hasScraped;
+
+    public PlanReporter(Optional<String> namespace, List<PlanManager> managers) {
+        this.managers = managers;
+        this.namespace = namespace;
+        this.executor = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder()
+                        .setDaemon(true)
+                        .setNameFormat("PlanReporterThread")
+                        .build()
+        );
+
+        this.hasScraped = new AtomicBoolean(false);
```

Comment:
Should we not just initialise the member directly since it's only used for testing?